rustls/src/crypto/mod.rs

```diff
@@ -186,6 +206,63 @@ pub struct CryptoProvider {
     pub key_provider: &'static dyn KeyProvider,
 }
 
+impl CryptoProvider {
+    /// Sets the supplied `CryptoProvider` as the default for this process.
+    ///
+    /// This can be called successfully at most once in any process execution.
+    ///
+    /// Call this early in your process to configure which provider is used for
+    /// the provider.  The configuration should happen before any use of
+    /// [`ClientConfig::builder()`] or [`ServerConfig::builder()`].
+    pub fn install_as_process_default(provider: impl Into<Arc<Self>>) -> Result<(), Arc<Self>> {
```

Comment:
I wonder if tying the process default APIs into the `CryptoProvider` value type is confusing? I don't find it all that intuitive because `install_as_process_default()` might as well take `self` (rather than `provider`).

Alternatively, we could have a separate unit type `DefaultProvider` with `set()` and `get()` methods, which feels like it would be more obvious. Or make `DEFAULT_PROVIDER` a `pub static` (potentially leaking once_cell into the public API, which may or may not be a big deal).

Reply:
Actually, just `self` works fine.

Reply:
This is much more obvious! I still find `install_as_process_default()` to be pretty verbose and would prefer a more concise spelling -- I don't think all of these tokens are necessary to make the meaning obvious.